src/clipping/weileratherton.jl

```diff
@@ -0,0 +1,408 @@
+            vertex_type = decide_direction(
+              Segment(clipped.left.point, point),
+              Segment(point, clipped.left.left.point),
+              Segment(clipping.point, clipping.left.point),
+              clippingsegment
+            )
+          end
+        elseif type(I) == CornerTouching
+          # When intersection is at the shared vertices of both the clipping and the clipped rings.
+          point = get(I)
+          if isapprox(point, clipped.left.point) && isapprox(point, clipping.left.point)
+            # Only applies if the intersection coincides with the middles of the currently observed
+            # vertices for both clipping and clipped rings.
+            vertex_type = decide_direction(
+              Segment(clipped.point, point),
+              Segment(point, clipped.left.left.point),
+              Segment(clipping.point, point),
+              Segment(point, clipping.left.left.point)
+            )
+          end
+        end
+        if !isnothing(vertex_type)
+          insertintersections!(clipping.left, clipped.left, point, vertex_type, entering)
+          intersected[k] = true
+        end
+
+        # Overlapping intersections might add up to two intersections, so handle separately.
+
+        if type(I) == Overlapping
+          for point in extrema(get(I))
+            # For both ends of the intersecting segment, check if it coincides with the middle of
+            # the obeserved vertices for clipped and clipping rings. If it does, attempt adding a
+            # point.
+
+            if isapprox(point, clipped.left.point)
+              vertex_type = decide_direction(
+                Segment(clipped.point, point),
+                Segment(point, clipped.left.left.point),
+                Segment(clipping.left.point, point),
+                Segment(point, clipping.left.left.point)
+              )
+              if !isnothing(vertex_type)
+                insertintersections!(clipping.left, clipped.left, point, vertex_type, entering)
+                intersected[k] = true
+              end
+            end
+
+            if isapprox(point, clipping.left.point)
+              vertex_type = decide_direction(
+                Segment(clipped.left.point, point),
+                Segment(point, clipped.left.left.point),
+                Segment(clipping.point, point),
+                Segment(point, clipping.left.left.point)
+              )
+              if !isnothing(vertex_type)
+                insertintersections!(clipping.left, clipped.left, point, vertex_type, entering)
+                intersected[k] = true
+              end
+            end
+          end
+        end
+
+        clipped = clipped.left
+        if clipped.left == startclipped.left
+          break
+        end
+      end
+    end
+
+    clipping = clipping.left
+    if clipping.left == startclipping.left
+      break
+    end
+  end
+
+  collectedrings::typeof(polyrings) = collectclipped(entering)
```

Comment:
```suggestion
  collectedrings = collectclipped(entering)
```

we don't use type annotations in algorithms

Reply:
This was there because it didn't work nicely with the PolyArea constructor without it. Now it works because the array is expanded in the constructor. Still, this might point to a need to add some corrections to the PolyArea constructors to work better with less specific arrays.